### ISSUE TITLE
processMailerCallback does not persist dnc entries

### ISSUE THREAD

**Issue by wittwerch** [mautic/mautic#2294]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | yes
| Feature request? | 
| Enhancement? | 

## Description:
I'm currently developing a mailer callback to handle bounce and complains from Amazon SES. The data from the callback ends in the processMailerCallback function, but never gets persisted.

Please have a look at: https://github.com/mautic/mautic/blob/7bcb68b9df0644e4f32b23dfa20347370d9cc6f2/app/bundles/EmailBundle/Model/EmailModel.php#L1478

There seems to be kind of a batch mechanism to lower the stress to the database, but imho the code is actually never increasing the counter.
This leads to the situation, that the expression on https://github.com/mautic/mautic/blob/7bcb68b9df0644e4f32b23dfa20347370d9cc6f2/app/bundles/EmailBundle/Model/EmailModel.php#L1550 is actually never true, and the entries are not persisted to the database.

I think even with a counter the code would actually not work. For example when processing 25 entries the last 5 would never be persisted.

## If a bug:

| Q   | A
| --- | ---
| Mautic version | v2.1.1-dev
| PHP version | 7.0.8-4+deb.sury.org~trusty+1

### Steps to reproduce:
1. Create a lead foo@bar.com
1.  Call processMailerCallback with the following array
````
Array ( [2] => Array ( [hashIds] => Array ( ) [emails] => Array ( [foo@bar.com] => smtp; 550 5.1.1 : Recipient address rejected: User unknown in virtual alias table ) ) [1] => Array ( [hashIds] => Array ( ) [emails] => Array ( ) ) )
```

=> Lead does not have an entry in lead_donotcontact

**Comment by alanhartless** [comment]
This was fixed in #2259 with https://github.com/mautic/mautic/pull/2259/files#diff-ca25c3d5ae42a898e6b89103a1acc8efR1492 but I'll move it into it's own PR so that we can get it into 2.1.1 as a bug fix.

**Comment by alanhartless** [comment]
Never mind on that last comment - I obviously was thinking something else once I started digging into this. But will fix this never-the-less :-)

**Comment by alanhartless** [comment]
Fixed in #2300

**Comment by wittwerch** [comment]
Thanks! Fix works and my pull request #2312 is on the way.. :)
